extensions-core/stats/src/main/java/io/druid/query/aggregation/teststats/PvaluefromZscorePostAggregator.java

```diff
@@ -23,98 +23,113 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonTypeName;
 import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterables;
 import com.google.common.collect.Sets;
+import io.druid.query.Queries;
 import io.druid.query.aggregation.AggregatorFactory;
 import io.druid.query.aggregation.PostAggregator;
+import io.druid.query.aggregation.post.ArithmeticPostAggregator;
 import io.druid.query.aggregation.post.PostAggregatorIds;
 import io.druid.query.cache.CacheKeyBuilder;
 import org.apache.commons.math3.distribution.NormalDistribution;
 
+import java.util.Collections;
 import java.util.Comparator;
 import java.util.Map;
 import java.util.Set;
 
-/**
- * Created by chunchen on 4/5/17.
- */
-@JsonTypeName("pvalue2tailedztest")
-public class PvaluefromZscorePostAggregator implements PostAggregator {
+@JsonTypeName("pvalue2tailedZtest")
+public class PvaluefromZscorePostAggregator implements PostAggregator
+{
   private final String name;
-  private final PostAggregator field;
+  private final PostAggregator zScore;
 
   @JsonCreator
   public PvaluefromZscorePostAggregator(
       @JsonProperty("name") String name,
-      @JsonProperty("field") PostAggregator field
-  ) {
+      @JsonProperty("zScore") PostAggregator zScore
+  )
+  {
     Preconditions.checkNotNull(name, "Must have a valid, non-null post-aggregator");
+    Preconditions.checkNotNull(zScore, "Must have a valid, non-null post-aggregator");
```

Comment:
Also it's better to check zScore is an intance of ZtestPostAggregator.

Reply:
One can calculate the z-score oneself without using ZtestPostAggregator.

Reply:
Actually, we want to keep it flexible and open.
For example, one can calculate it's own z-score for example if needed, and just feed its zscore to calculate p-value.

Reply:
Ah, ok. It makes sense.